### DISCUSSION TITLE
Custom metric for evaluating best_model

### DISCUSSION THREAD

**Discussion by Pacman1984** [ThilinaRajapakse/simpletransformers#911] [Q&A] [answered]
Great Job with this Package! It reduces the code to train a model on average by ~80% and is more readable than plain pytorch and transformers.

One Question: How to save best_model based on custom metric (like F1_score) instead of eval_loss?

Best

**Comment by Timliuzhaolu** [comment]
Yeah, I have got a similar question. How to compute other metrics in the evaluation process? Right now classification is only outputting mcc and multilabel classification is only outputting LRAP. How can we use metrics like accuracy, precision recall etc. Maybe with the compute_metrics function? Or maybe there is a way to output tn, tp, fn, fp so that we can calculate manually?

Thank you so much!

Tim

**Reply by Pacman1984** [reply]
Hi,

this is possible by giving the scorer function as arguments to the train_model() and eval_model() methods

Example:
https://simpletransformers.ai/docs/tips-and-tricks/#additional-evaluation-metrics

Best

**Reply by Timliuzhaolu** [reply]
That is terrific! Thanks for the reply!

**Comment by ThilinaRajapakse** [comment, marked as answer]
The `early_stopping_metric` and `early_stopping_metric_minimize` args can be used to do this. Note that using these args won't turn on early stopping unless you set `use_early_stopping`. Admittedly, the argument names are a little misleading in this case though.
